miio/tests/test_vacuums.py

```diff
@@ -0,0 +1,55 @@
+"""Test of vacuum devices."""
+from collections.abc import Iterable
+from typing import List, Sequence, Tuple, Type
+
+import pytest
+
+from miio.device import Device
+from miio.integrations.vacuum.roborock.vacuum import ROCKROBO_V1
+from miio.interfaces import VacuumInterface
+
+# list of all supported vacuum classes
+VACUUM_CLASSES: Tuple[Type[VacuumInterface], ...] = tuple(
+    cl for cl in VacuumInterface.__subclasses__()  # type: ignore
+)
+
+
+def _all_vacuum_models() -> Sequence[Tuple[Type[Device], str]]:
+    """:return: list of tuples with supported vacuum models with corresponding class"""
+    result: List[Tuple[Type[Device], str]] = []
+    for cls in VACUUM_CLASSES:
+        assert issubclass(cls, Device)
+        vacuum_models = cls.supported_models
+        assert isinstance(vacuum_models, Iterable)
+        for model in vacuum_models:
+            result.append((cls, model))
+    return result  # type: ignore
+
+
+@pytest.mark.parametrize("cls, model", _all_vacuum_models())
+def test_vacuum_fan_speed_presets(cls: Type[Device], model: str) -> None:
+    """Test method VacuumInterface.fan_speed_presets()"""
+    if model == ROCKROBO_V1:
+        return  # this model cannot be tested because presets depends on firmware
```

Comment:
Mocking the model check would allow verifying that at least the default case is working and the API is implemented. After all, all implementations should follow the API no matter how the fan speed presets are obtained.

Reply:
I'll do thin in next PR.